src/hashTable.js

```diff
@@ -0,0 +1,60 @@
+export default class HashTable {
+  constructor() {
+    this.storage = []
+    this._length = 0
+    this.max = 10
+  }
+
+  put = (key, value) => {
+    if(typeof(key) === "undefined"){
+      throw("Cannot insert with undefined key!")
+    }
+    const hashIndex = this.hash(key, this.max)
+    this.storage[hashIndex] = value
+    this._length += 1
+  }
+
+  get = (key) => {
+    const hashIndex = this.hash(key, this.max)
```

Comment:
This would not work if you have key collision (when different keys yield the same hash) - this is very likely to happen in smaller key spaces.

In order to fix that possible problem, we can use a strategy called [separate chaining](http://algs4.cs.princeton.edu/34hash/), where each entry in the hash table is a linked list.  In order to retrieve, we hash the key for the index, then search through the linked list at that index for the `Node` with the matching key.

Without separate chaining, we need to develop a "probing" strategy - an algorithm that allows us to traverse the hash table following a specific protocol in order to store key value pairs whose hashed index is already taken (linear and quadratic probing).